### PR TITLE
Add experimental interop mobile results view

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
This change pulls mobile results data from the [interop-results repo](https://github.com/jgraham/interop-results) and uses it to display a new mobile results view on the Interop dashboard. The mobile results view can only be seen by toggling a flag in `/flags`.

https://github.com/user-attachments/assets/afeef440-e2f3-4727-8e13-4132cbd028dc

### CODE REVIEW

**Comment by DanielRyanSmith** [comment]
It looks like the linter is failing due to an issue unrelated to this change.

```
api/query/cache/index/index.go:210:18: G115: integer overflow conversion uint64 -> int (gosec)
		shardIdx := int(t.testID % numShardsU64)
		               ^
api/query/cache/backfill/backfill.go:120:14: G115: integer overflow conversion uint64 -> int (gosec)
	limit := int(maxBytes/bytesPerRun) / len(shared.GetDefaultProducts())
```

https://github.com/securego/gosec/pull/1149

**Comment by KyleJu** [comment]
> It looks like the linter is failing due to an issue unrelated to this change.
> 
> ```
> api/query/cache/index/index.go:210:18: G115: integer overflow conversion uint64 -> int (gosec)
> 		shardIdx := int(t.testID % numShardsU64)
> 		               ^
> api/query/cache/backfill/backfill.go:120:14: G115: integer overflow conversion uint64 -> int (gosec)
> 	limit := int(maxBytes/bytesPerRun) / len(shared.GetDefaultProducts())
> ```
> 
> [securego/gosec#1149](https://github.com/securego/gosec/pull/1149)

I will get to it shortly

**Comment by DanielRyanSmith** [comment]
View the staging deployment here: https://2024-08-25-mobile-resu-dot-wptdashboard-staging.uk.r.appspot.com/interop-2024?mobileView

It might be worth changing this view in some way to be even more explicit that these are mobile results and experimental results.

**Comment by DanielRyanSmith** [comment]
Some odd logic has been added in a commit that can be reverted later:

https://github.com/web-platform-tests/wpt.fyi/pull/3959/commits/9068283c90fcd151f607a13303642b65e2d2c8cd 

Once we have Safari results to populate the page

File: webapp/components/interop-dashboard.js [hunk 1]
```diff
@@ -578,7 +616,16 @@ class InteropDashboard extends PolymerElement {
     const params = (new URL(document.location)).searchParams;
 
     this.stable = params.get('stable') !== null;
-    this.dataManager = new InteropDataManager(this.year);
+    this.isMobileScoresView = params.get('mobileView') !== null && this.showMobileScoresView;
```
Comment (KyleJu) [inline webapp/components/interop-dashboard.js:619]: See my comments below. The param should be separated using -, `mobile-view`

File: webapp/components/interop-data-manager.js [hunk 2]
```diff
@@ -245,9 +290,12 @@ class InteropDataManager {
         browserVersions[browserIdx].push(version);
 
         let testScore = 0.0;
+        // Mobile csv does not have an Interop version column to account for.
+        let versionOffset = (this.isMobileScoresView && browserName === 'Interop') ? 0 : 1;
```
Comment (KyleJu) [inline webapp/components/interop-data-manager.js:294]:
```suggestion
        const versionOffset = (this.isMobileScoresView && browserName === 'Interop') ? 0 : 1;
```

File: webapp/interop_handler.go [hunk 3]
```diff
@@ -36,6 +37,18 @@ func interopHandler(w http.ResponseWriter, r *http.Request) {
...
+	q := r.URL.Query()
+	isMobileView, err := shared.ParseBooleanParam(q, "mobileView")
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	_, isValidMobileYear := validMobileYears[year];
+	if isMobileView != nil && !isValidMobileYear {
+		year = defaultRedirectYear
+		needsRedirect = true
```
Comment (KyleJu) [inline webapp/interop_handler.go:49]: minor detail: should it redirect to interop-2024? or interop-2024?mobile-view?

**Review by KyleJu — APPROVED** [review]
LGTM with some minor comments

File: webapp/interop_handler.go [hunk 4]
```diff
@@ -36,6 +37,18 @@ func interopHandler(w http.ResponseWriter, r *http.Request) {
 		needsRedirect = true
 	}
 
+	q := r.URL.Query()
+	isMobileView, err := shared.ParseBooleanParam(q, "mobileView")
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply webapp/interop_handler.go:41]: Good call - I'll change to use that syntax

File: webapp/components/wpt-flags.js [hunk 5]
```diff
@@ -258,6 +259,11 @@ class WPTFlagsEditor extends FlagsEditorClass(/*environmentFlags*/ false) {
         Enable Browser Specific Failures graph
       </paper-checkbox>
     </paper-item>
+    <paper-item>
+      <paper-checkbox checked="{{showMobileScoresView}}">
+        Allow mobile results view on Interop dashboard
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply webapp/components/wpt-flags.js:264]: I like it better. Changed!

File: webapp/components/interop-dashboard.js [hunk 6]
```diff
@@ -292,14 +301,24 @@ class InteropDashboard extends PolymerElement {
...
+          <interop-summary
+            year="[[year]]"
+            data-manager="[[dataManager]]"
+            scores="[[scores]]"
+            stable="[[stable]]"
+            is-mobile-scores-view="[[isMobileScoresView]]">
+          </interop-summary>
         </div>
         <div class="grid-item grid-item-description">
           <p>Interop [[year]] is a cross-browser effort to improve the interoperability of the web —
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply webapp/components/interop-dashboard.js:324]: This is an [em dash](https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use) and was chosen to use by the team that wrote up this dashboard description

File: webapp/interop_handler.go (hunk 3 above)
Comment (KyleJu) [reply webapp/interop_handler.go:49]: I don't think that is the case testing on the staging deployment - https://2024-08-25-mobile-resu-dot-wptdashboard-staging.uk.r.appspot.com/interop-2023?mobile-view

Comment (KyleJu) [reply webapp/interop_handler.go:49]: nm it worked! The staging deployment was not up-to-date, https://2024-08-25-mobile-resu-dot-wptdashboard-staging.uk.r.appspot.com/interop-2023?mobile-view. Thanks!

File: webapp/components/interop-dashboard.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (KyleJu) [reply webapp/components/interop-dashboard.js:324]: Aha thanks!

**Pull request merged**